### DISCUSSION TITLE
bug: Reverse Proxy breaks Microsoft OAuth

### DISCUSSION THREAD

**Discussion by YetheSamartaka** [open-webui/open-webui#3947] [General]
# Bug Report

## Description

**Bug Summary:**
 I have WebUI running on Ubuntu 22.04 via Docker, latest version from GitHub and I have reverse proxy setup. So far, everything was pretty much working alright, but when I wanted to add Microsoft OAuth, I've stumbled upon an issue where the redirect URI that is generated is HTTP instead of HTTPS one. 

**Expected Behavior:**
HTTPS Link is generated.

**Actual Behavior:**
HTTP Link is generated.

## Environment

- **Open WebUI Version:** [0.3.9]

- **Operating System:** [Ubuntu 20.04]

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.

## Installation Method

Dockerfile

## Additional Information

I found Discussion on the GitHub mentioning probably the similar issue https://github.com/open-webui/open-webui/discussions/3727
Where yusiwen is commenting the backend/start.sh, line 58 as of this date that is missing the --proxy-headers argument. However, I've put it on my local fork there, rebuilt the docker image and I still have the issue. How can I proceed further, please?

**Comment by JScheitelBEPC** [comment]
Running into the same issue - when setting up the callback URL for the application on the Microsoft side, it will not accept non-ssl callback URL...  When you secure Open-WebUI with SSL (in our case using stunnel) - we get AADSTS50011 error stating that the http:// callback passed in the request does not match the https:// callback configured for the application.  Willing to test if anyone has ideas on workarounds...  thanks!

**Comment by cdgco** [comment]
Experiencing the same thing with the Microsoft OAuth provider, but I found that there are [`MICROSOFT_REDIRECT_URI`](https://github.com/open-webui/open-webui/blob/99d10d1189452ad49fcace219e9c90ae65906cd1/backend/config.py#L382) and [`OPENID_REDIRECT_URI`](https://github.com/open-webui/open-webui/blob/99d10d1189452ad49fcace219e9c90ae65906cd1/backend/config.py#L406) variables that aren't listed in the documentation. Setting the `MICROSOFT_REDIRECT_URI` variable to the `https://{URL}/oauth/microsoft/callback` URL worked for me.

**Comment by JScheitelBEPC** [comment]
thanks @cdgco - I was able to get it working too using MICROSOFT_REDIRECT_URI and not sure if it was needed or not, but also set WEBUI_URL to an https value.

**Comment by Naxter** [comment]
Would you guys mind giving a short overview on how your setup looks like? I am struggling to get it running at the moment. Would highly appreciate it.

**Reply by JScheitelBEPC** [reply]
It kind of depends on your chosen configuration, but to fix the issue addressed here, we are basically starting the container with this command:

`docker run -d -p 127.0.0.1:3000:8080 -e OLLAMA_BASE_URL=http://xxx.xx.xxx.xxx:11434 -v open-webui:/app/backend/data -e MICROSOFT_CLIENT_ID=your_app_id -e MICROSOFT_CLIENT_SECRET=your_app_secret_value -e MICROSOFT_CLIENT_TENANT_ID=your_tenant_id -e MICROSOFT_REDIRECT_URI=https://your.url.com/oauth/microsoft/callback -e ENABLE_OAUTH_SIGNUP=true -e WEBUI_URL=https://your.url.com -e DEFAULT_USER_ROLE=user  -e GLOBAL_LOG_LEVEL=DEBUG --name open-webui --restart always ghcr.io/open-webui/open-webui:main`

That being said - we are using docker and then using stunnel to reverse proxy with a enterprise issued cert.  Try the discord if you want specific help with your configuration.

**Reply by Naxter** [reply]
Appreciate your answer. Thank you.
